cfme/storage/volume.py

```diff
@@ -121,53 +167,117 @@ def is_displayed(self):
     form = View.nested(VolumeAddForm)
 
 
+class VolumeCollection(Navigatable):
+    """Collection object for the :py:class:'cfme.storage.volume.Volume'. """
+    def __init__(self, appliance=None):
+        self.appliance = appliance
+
+    def instantiate(self, name, provider):
+        return Volume(name, provider, collection=self)
+
+    def delete(self, *volumes):
+        """Delete one or more Volumes from list of Volumes
+
+        Args:
+            list of the 'cfme.storage.volume.Volume' objects
+        """
+
+        view = navigate_to(self, 'All')
+
+        if view.entities.get_all():
+            for volume in volumes:
+                try:
+                    view.entities.get_entity(volume.name).check()
+                except:
+                    raise VolumeNotFound("Volume {} not found".format(volume.name))
+
+            view.toolbar.configuration.item_select('Delete selected Cloud Volumes',
+                                                   handle_alert=True)
+
+            for volume in volumes:
+                volume.wait_for_disappear()
+        else:
+            raise VolumeNotFound('No Cloud Volume for Deletion')
+
+
 class Volume(Navigatable):
     # Navigation menu option
     nav = VersionPick({
         Version.lowest(): ['Storage', 'Volumes'],
         '5.8': ['Storage', 'Block Storage', 'Volumes']})
 
-    def __init__(self, name, provider, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
+    def __init__(self, name, provider, collection=None):
         self.name = name
         # TODO add storage provider parameter, needed for accurate details nav
         # the storage providers have different names then cloud providers
         # https://bugzilla.redhat.com/show_bug.cgi?id=1455270
         self.provider = provider
+        self.collection = collection or VolumeCollection()
+        Navigatable.__init__(self, appliance=self.collection.appliance)
+
+    def wait_for_disappear(self, timeout=300):
+        def refresh():
+            self.provider.refresh_provider_relationships()
+            self.browser.refresh()
+
+        try:
+            return wait_for(lambda: self.exists,
+                            fail_condition=True,
+                            timeout=timeout,
+                            message='Wait for cloud Volume to disappear',
+                            delay=20,
+                            fail_func=refresh)
+        except TimedOutError:
+            logger.error('Timed out waiting for Volume to disappear, continuing')
+
+    def delete(self, wait=True):
+        """Delete the Volume"""
+
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Delete this Cloud Volume', handle_alert=True)
+
+        result = view.entities.flash.assert_success_message(
+            'Delete initiated for 1 Cloud Volume.')
+        if wait:
+            result = self.wait_for_disappear(500)
+        return result
```

Comment:
Why do you return result? Is it used somewhere?

Reply:
No use, As crude test not there, I return it for crud test if someone finds a need. I took reference from tenant file. But I am going to remove it now.